pytorch_lightning/utilities/dataloader_preparation.py

```diff
@@ -0,0 +1,163 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
Why does this code need to be moved into an utility?

IMO it's better to have these things together and we don't want to bloat the utilities.

Additionally, moving it messes up the git history so unless there's is a good reason, I wouldn't do it

Reply:
data loading is quite complex and I think grouping the utilities dataloader would help to understand the code better.

Reply:
I'd argue that grouping the utilities in the same file `data_loading.py` is more readable since you avoid the file jump. It also makes patching and subclassing easier.

Reply:
Fine ! Done !